backend/python/app/middlewares/user.py

```diff
@@ -0,0 +1,29 @@
+from flask import current_app, jsonify, request
+from functools import wraps
+
+from ..resources.create_user_dto import CreateUserDTO
+from ..resources.update_user_dto import UpdateUserDTO
+
+def validate_create_user(clazz):
+    """
+    Determine if request is authorized based on access_token validity and role of client
+
+    :param roles: the set of authorized roles to check for
+    :type roles: {str}
+    """
+
+    def validate_dto(api_func):
+        @wraps(api_func)
+        def wrapper(*args, **kwargs):
+            user = eval(clazz)(**request.json)
```

Comment:
Would calling the constructor here fail if `request.json` doesn't contain enough parameters (or if it doesn't contain the correct parameters)?

I really like this design though, very nice and reusable, great work! To make this work the DTO class's constructor likely needs to be modified to take `**kwargs` (keyword arguments). Only the constructor needs to be changed, how the constructor is called will not. For example, `CreateUserDTO`:

```python
# current constructor, this will fail if the not-yet-validated json contains an unexpected field or if its missing a field
def __init__(self, first_name, last_name, email, role, password):
    self.first_name = first_name
    ...

# with kwargs
def __init__(self, **kwargs):
    # if first_name is not provided, self.first_name will be assigned None, will check in validate method
    # kwargs is treated as a dict here
    self.first_name = kwargs.get("first_name")
    ...
```

Reply:
I've changed it to `first_name=None, last_name=None` in the function's parameters to help give argument hints if someone is trying to use the constructor. Let me know if you still prefer I use `**kwrgs` instead

Reply:
Yep, I think `kwargs` is still needed just because it's possible for the user to supply unsupported fields, e.g. constructing a `CreateUserDTO` from this request will result in an exception since `foo` is not a constructor argument

```json
{
    "foo": "bar"
}
```